src/webrtc/call.ts

```diff
@@ -697,7 +712,14 @@ export class MatrixCall extends EventEmitter {
             this.waitForLocalAVStream = true;
 
             try {
-                const mediaStream = await navigator.mediaDevices.getUserMedia(constraints);
+                let mediaStream: MediaStream;
+
+                if (this.type === CallType.Voice) {
+                    mediaStream = await this.client.getLocalAudioStream();
```

Comment:
If we remove the public methods for getting local audio stream, we need to add some private method/state in its place.

Reply:
Maybe `localAVStream` / `localAVStreamPromise` could just be moved to a variable within this module? We only ever want to reference it from the call class. Also then we don't need to export `getUserMediaContraints`

Reply:
Spent some time trying to figure out a better way to do this and I couldn't come up with a great solution. We actually want to call `getLocalAudioStream` from both `call.ts` and `groupCall.ts` so it can't be local to `call.ts`. I guess I need to know if it's alright to leave these as public methods or not.

Reply:
As mentioned above the [`MediaHandler`](https://github.com/SimonBrandner/matrix-js-sdk/commit/a1f73eee86aff8b5b3c667573b44942c3c4e23f9) from https://github.com/matrix-org/matrix-js-sdk/pull/1911 feels like the way to go to me

Reply:
I moved everything to use #1911 and #1928